prover/vk_setup_data_generator_server_fri/src/keystore.rs

```diff
@@ -44,7 +44,18 @@ pub struct Keystore {
 }
 
 fn get_base_path() -> PathBuf {
-    core_workspace_dir_or_current_dir().join("prover/vk_setup_data_generator_server_fri/data")
+    let path = core_workspace_dir_or_current_dir();
+
+    let new_path = path.join("prover/vk_setup_data_generator_server_fri/data");
+    if new_path.exists() {
+        return new_path;
+    }
+
+    let mut components = path.components();
```

Comment:
please add a comment what is happening here.

Reply:
This is a kludge for setup data generator to work in CI. Unfortunately I fail to understand how it's supposed to work, it seems in production it also doesn't work as expected and relies on the fact that we run binaries from some path.